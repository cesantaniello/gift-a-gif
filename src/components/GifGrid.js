import React from 'react'

export const GifGrid = ({category}) => {

    const getGifs = async() => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=batman&limit=10&api_key=AnCjC7Tgzj8tJO9Jj91ImRnw3mdlMsqp'
        const resp = await fetch(url);
        const {data} = await resp.json();



        console.log(data);
    }

    getGifs();

    return (
        <div>
            <h3>{category}</h3>
        </div>
    )
}
