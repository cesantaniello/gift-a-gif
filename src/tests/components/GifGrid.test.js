import React from 'react';
import { shallow } from 'enzyme';
import {GifGrid} from '../../components/GifGrid';


describe('Pruebas en <GifGrid />', () => {
    test('Debe de mostrarse correctamente', () => {
        const category = 'One Punch';
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {
        const category = 'One Punch';
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/imagen.jpg',
            title: 'Imagen'
        }];
        const wrapper = shallow(<GifGrid category={category} gifs={gifs} />);
        expect(wrapper).toMatchSnapshot();
    })
})