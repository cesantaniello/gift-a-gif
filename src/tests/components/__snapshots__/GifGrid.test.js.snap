// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
